src/App.tsx

```diff
@@ -1,24 +1,244 @@
+        setTypeError(TypeError.Unexpected);
+        setTextError(TypeError.Unexpected);
+      });
+  }, []);
+
+  useEffect(() => {
+    const timerId = setTimeout(setTypeError, 3000, '');
+
+    return () => clearTimeout(timerId);
+  }, [typeError]);
+
+  switch (filterByStatus) {
+    case Status.Active:
+      visibleTodos = activeTodos;
+      break;
+
+    case Status.Completed:
+      visibleTodos = completedTodos;
+      break;
+
+    default:
+      visibleTodos = todos;
+      break;
+  }
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <TodoAppHeader
+          allTodosCompleted={areAllTodosCompleted}
+          disableInput={disableInput}
+          titleTodo={titleTodo}
+          setTitleTodo={setTitleTodo}
+          addHandler={addHandler}
+          onToggleAll={toggleAllHandler}
+        />
+
+        {todos.length > 0 && (
+          <>
+            <section className="todoapp__main">
+              <TransitionGroup>
+                {visibleTodos.map(todo => (
+                  <CSSTransition
+                    key={todo.id}
+                    timeout={500}
+                    classNames="item"
+                  >
+                    <TodoAppTodo
+                      todo={todo}
+                      deleteHandler={deleteHandler}
+                      isProcessed={idTodosLoading.includes(todo.id)}
+                      onUpdate={updateHandler}
+                    />
+                  </CSSTransition>
+                ))}
+                {tempTodo && (
+                  <CSSTransition
+                    key={0}
+                    timeout={500}
+                    classNames="temp-item"
+                  >
+                    <TodoAppTodo
+                      todo={tempTodo}
+                      isProcessed
+                    />
+                  </CSSTransition>
+                )}
+              </TransitionGroup>
+            </section>
+
+            <TodoAppFooter
+              itemsLeft={activeTodos.length}
+              filterByStatus={filterByStatus}
+              setFilterByStatus={setFilterByStatus}
+              areTodosCompleted={completedTodos.length > 0}
+              clearCompleted={deleteHandler}
+            />
+          </>
+        )}
+      </div>
+
+      <div
+        className={classNames(
+          'notification',
+          'is-danger',
+          'is-light',
+          'has-text-weight-normal',
+          { hidden: typeError.length === 0 },
```

Comment:
```suggestion
          { hidden: !typeError },
```
empty string: `''` === false